src/test/kotlin/systems/ajax/malov/stockanalyzer/repository/impl/MongoStockRecordRepositoryTest.kt

```diff
@@ -19,32 +18,37 @@ class MongoStockRecordRepositoryTest : AbstractMongoIntegrationTest {
 
     @Test
     fun `insertAll fun inserts all stocks records and retrieves inserted stock records ids`() {
-        val listOfUnsavedStockRecords = listOf(unsavedStockRecord())
+        val expected = listOf(unsavedStockRecord())
 
-        val actual = mongoStockRecordRepository.insertAll(listOfUnsavedStockRecords)
+        val actual = mongoStockRecordRepository.insertAll(expected)
 
-        assertEquals(listOfUnsavedStockRecords.size, actual.size)
-        assertEquals(listOfUnsavedStockRecords.size, actual.count { it.id != null })
+        StepVerifier.create(actual)
+            .expectNextMatches { expected[0].id != null }
+            .verifyComplete()
     }
 
     @Test
     fun `getAllStockSymbols retrieves all stock symbols`() {
-        val listOfUnsavedStocks = listOf(unsavedStockRecord())
+        val listOfUnsavedStocks = listOf(unsavedStockRecord().copy(symbol = "AJAX"))
         mongoStockRecordRepository.insertAll(listOfUnsavedStocks)
+            .subscribe()
 
         val actual = mongoStockRecordRepository.findAllStockSymbols()
 
-        assertTrue(actual.toSet().containsAll(listOfUnsavedStocks.map { it.symbol }))
+        StepVerifier.create(actual)
+            .expectNextMatches { result -> result.toSet().containsAll(listOfUnsavedStocks.map { it.symbol }) }
```

Comment:
pls, review all tests where u use `expectNextMatches`

Reply:
What do you mean exactly?